Customizable grid size

f-i-l-i-p opened the pull request:
This extends the current pixel grid to be more customizable by allowing the user to specify the grid size as described in #1028. The original functionality of the pixel grid is still available be setting the grid width and height to 1px.

https://github.com/user-attachments/assets/ccd5470e-9f90-473a-b528-ba03c5c5b4ad

cameronwhite commented on Pinta.Gui.Widgets/Widgets/Canvas/CanvasRenderer.cs:
```diff
@@ -103,13 +103,28 @@ public void Render (
-		if (enable_pixel_grid && PintaCore.Actions.View.PixelGrid.Value && scale_factor.Ratio <= 0.5d)
+		// Draw the grid if it's enabled and we're zoomed in far enough
+		const int minGridLineDistance = 5;
+		if (enable_grid && PintaCore.CanvasGrid.ShowGrid && GetMinGridLineDistance () >= minGridLineDistance)
```
Could the canvas renderer receive an `ICanvasGridService` to eliminate the global PintaCore dependency?
This could actually also eliminate the `enable_grid` member too if you use a nullable `ICanvasGridService?` to indicate that the canvas grid should never be displayed (e.g. for the preview images in the layers list view widget)

cameronwhite commented on Pinta/Actions/View/EditCanvasGridAction.cs:
```diff
@@ -0,0 +1,50 @@
+	{
+		this.view = view;
+		this.chrome = chrome;
+		canvas_grid = canvasGrid;
```
Minor nitpick - the style is inconsistent here with using a mix of `this.foo = ...` and just `foo = ...`

cameronwhite commented on Pinta/Dialogs/CanvasGridSettingsDialog.cs:
```diff
@@ -0,0 +1,110 @@
+
+		grid.Attach (show_grid_checkbox, 0, 0, 2, 1);
+
+		grid.Attach (CreateLabel ("Width:", Align.End), 0, 1, 1, 1);
```
The call to `Translations.GetString()` actually needs to happen here around the string literal, e.g. `Translations.GetString("Width:")`, rather than inside the `CreateLabel()` method - otherwise these strings won't be picked up with the automated tool that finds all the translatable strings (https://github.com/PintaProject/Pinta/wiki/Translations)

(Same for the `Height:` label below)

cameronwhite commented on Pinta/Dialogs/CanvasGridSettingsDialog.cs:
```diff
@@ -0,0 +1,110 @@
+		mainVbox.SetOrientation (Orientation.Vertical);
+		mainVbox.Append (grid);
+
+		Title = Translations.GetString ("Grid Settings");
```
The dialog title could be `Canvas Grid Settings` to match the naming of the menu label suggested in my other comment?

cameronwhite reviewed (commented):
This looks great, thanks for working on this! Just left several minor suggestions

f-i-l-i-p commented:
> This looks great, thanks for working on this! Just left several minor suggestions

Thanks for the feedback! I think that I have addressed all of your points
